app/src/main/java/com/team22/soundary/core/auth/TokenAuthenticator.kt

```diff
@@ -0,0 +1,28 @@
+package com.team22.soundary.core.auth
+
+import com.team22.soundary.core.domain.TokenRepository
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import okhttp3.Authenticator
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.Route
+import javax.inject.Inject
+
+class TokenAuthenticator @Inject constructor(
+    private val tokenRepository: TokenRepository
+) : Authenticator {
+
+    override fun authenticate(route: Route?, response: Response): Request? {
+        val token : String = runBlocking {
+            tokenRepository.getRefreshToken().first().getOrNull()
+        } ?: ""
+
+        return Request.Builder().addHeader(AUTH, TOKEN_PREFIX+token).build()
```

Comment:
p5; TOKEN_PREFIX+token 보다 조금 더 코틀린 스럽게 처리해보죠!
```suggestion
        return Request.Builder().addHeader(AUTH, "$TOKEN_PREFIX$token").build()
```

Reply:
p3; 만약 token 이 `""` 일 때는굳이 요청하지 않아도 401 에러가 나올 것 같은데요, Request 를 하지 않고 response 를 직접 생성해서 반환해도 좋을 것 같아요.

Reply:
token이 ""일 경우 null 반환하도록 하여 추가적인 request 안보내도록 수정하였습니다!!